InterfacePython/testBasicIO.py

```diff
@@ -410,6 +410,59 @@ def checkReadWriteMisc(self):
                 gridB.read("testSave")
                 ttc.compareGrids(gridA, gridB)
 
+    def checkReadWriteCustomTabulated(self):
+        '''
+        Test reading and writing of the CustomTabulated class.
+        '''
+        rng = np.random.default_rng(777);
+        description = "testCT"
+        def create_nodes(j, in_rng):
+            nodes = in_rng.random(j) * 2.0 - 1.0
+            nodes.sort()
+            return nodes
+        def create_weights(j, in_rng):
+            weights = in_rng.random(j)
+            weights = 2.0 * weights / weights.sum()
+            return weights
+        # Read and write from explicitly given data.
+        for i in range(4):
+            num_levels = i
+            num_nodes = np.array([3*(j+1) for j in range(i)])
+            precision = np.array([2*(j+1)-1 for j in range(i)])
+            nodes = [create_nodes(j, rng) for j in num_nodes]
+            weights = [create_weights(j, rng) for j in num_nodes]
+            ctA = TasmanianSG.makeCustomTabulatedFromData(num_levels, num_nodes, precision, nodes, weights, description)
+            ctB = TasmanianSG.CustomTabulated()
+            ctA.write("testSave")
+            ctB.read("testSave")
+            ttc.compareCustomTabulated(ctA, ctB)
+        # Read and write from a file.
+        ctA = TasmanianSG.makeCustomTabulatedFromFile(tdata.sGaussPattersonTableFile)
+        ctB = TasmanianSG.CustomTabulated()
+        ctA.write("testSave")
+        ctB.read("testSave")
+        ttc.compareCustomTabulated(ctA, ctB)
+        # Test an error message from wrong read.
+        try:
+            ctB.read("Test_If_Bogus_Filename_Produces_an_Error")
+        except TasmanianSG.TasmanianInputError as TSGError:
+            TSGError.bShowOnExit = False
+            self.assertEqual(TSGError.sVariable, "sFilename", "Reading a bogus file properly failed, but the error information is wrong.")
+
+    def checkGlobalGridCustom(self):
+        '''
+        Test makeGlobalGridCustom(), which creates a grid from a CustomTabulated instance.
+        '''
+        ct = TasmanianSG.makeCustomTabulatedFromFile(tdata.sGaussPattersonTableFile)
+        gridA1 = TasmanianSG.TasmanianSparseGrid()
```

Comment:
Use a more logical order of operations:
* initialize gridA1, you can also use `TasmanianSG.makeGlobalGrid()` instead of making an empty grid and then overwritting
* initialize `ct`
* use `ct` to initialize `gridB1`

Right now you start with `ct` but you don't use it until later which breaks the flow when someone is reading the code.

Reply:
Changed, and renamed `gridA1` (resp. `gridB1`) to `gridA` (resp. `gridB`) to be consistent with the rest of the code.